examples/reduce.c

```diff
@@ -0,0 +1,339 @@
+	ptl_handle_md_t mdh;
+	char msg[PTL_EV_STR_SIZE];
+	ptl_event_t ev;
+	ptl_handle_eq_t eqh;
+	ptl_index_t pti0;
+	ptl_index_t pti1;
+
+	int ret = PtlEQAlloc(nih, 10, &eqh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+
+	ret = PtlPTAlloc(nih, PTL_PT_FLOWCTRL, eqh, PTL_PT_ANY, &pti0);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlPTAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+
+	/* init me0 */
+
+	ptl_me_t me0 = { .start = ranks[rank].data,
+			 .length = ranks[rank].size,
+			 .ct_handle = PTL_CT_NONE,
+			 .uid = PTL_UID_ANY,
+			 .options = PTL_ME_OP_PUT,
+			 .match_id = PTL_NID_ANY,
+			 .match_bits = 0,
+			 .ignore_bits = 0,
+			 .min_free = 0 };
+
+	ret = PtlLEAppend(nih, pti0, &me0, PTL_PRIORITY_LIST, NULL, &leh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlLEAppend failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+
+	ret = PtlEQWait(eqh, &ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+	PtlEvToStr(0, &ev, msg);
+	printf("Event : %s", msg);
+
+	ptl_md_t md = { .start = ranks[rank].rank,
+			.length = ranks[rank].size,
+			.options = 0,
+			.eq_handle = eqh,
+			.ct_handle = PTL_CT_NONE };
+
+	ret = PtlMDBind(nih, &md, &mdh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlMDBind failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+
+	/* sending the content of md to me0 */
+	ret = PtlAtomic(mdh, 0, ranks[rank].size, PTL_ACK_REQ, id, pti0, 0, 0, NULL, 0, PTL_SUM,
+			PTL_UINT32_T);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlAtomic failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	} else {
+		ret = PtlCTInc(ranks[rank].counter, increment);
+		if (ret != PTL_OK) {
+			fprintf(stderr, "PtlCTInc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+			return 1;
+		}
+	}
+
+	ret = PtlEQWait(eqh, &ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+	PtlEvToStr(0, &ev, msg);
+	printf("Event : %s", msg);
+
+	ret = PtlEQWait(eqh, &ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+	PtlEvToStr(0, &ev, msg);
+	printf("Event : %s", msg);
+
+	ret = PtlEQWait(eqh, &ev);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+	PtlEvToStr(0, &ev, msg);
+	printf("Event : %s", msg);
+
+	if (rank < 2) {
+		side = "left";
+	} else {
+		side = "right";
+	}
+	int *num0 = me0.start;
+	printf("content of me0 %s: {%d,%d,%d} \n", side, num0[0], num0[1], num0[2]);
+
+	/* if we have completed me0 with both md */
+	ret = PtlEQAlloc(nih, 10, &eqh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+
+	ret = PtlPTAlloc(nih, PTL_PT_FLOWCTRL, eqh, PTL_PT_ANY, &pti1);

```

Comment:
I don't think we need PTL_PT_FLOWCTRL, and I think we should choose not PTL_PT_ANY, but our rank index, that would be cleared